### PR TITLE
wr-265: Fix the layout

### PR DESCRIPTION
**Pull request by PablloZz**
1.  Fixed the feed and my articles tabs layout:

image
image
image

2. Added word breaking for very long names:

image
3. Expanded the text in the article page:

image

4. Fixed genre stats appearing:

image
image

5. Fixed the password fields:
image

6. Added placeholder if there aren't any articles yet:
image

### CODE REVIEW

**Review by andanf-e — COMMENTED** [review]
Let's move empty book image to the placeholder center
image
And please move close button under modal container padding
image
And also when article does not have prompt - let's make single genre not stretching on full width
image

File: frontend/src/libs/helpers/check-window-breakpoint/check-window-breakpoint.helper.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import {
+  WindowBreakpoint,
+  WindowBreakpointThreshold,
+} from '~/libs/enums/enums.js';
+import { type ValueOf } from '~/libs/types/types.js';
+
+const checkWindowBreakpoint = (
```
Comment (andanf-e) [inline frontend/src/libs/helpers/check-window-breakpoint/check-window-breakpoint.helper.ts]:
```suggestion
const checkIsPassingWindowBreakpoint = (
```

File: frontend/src/libs/hooks/use-get-window-dimensions/use-get-window-dimensions.hook.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import { useEffect, useState } from '../hooks.js';
+
+type ReturnType = {
+  width: number;
+  height: number;
+};
+
+const useGetWindowDimensions = (): ReturnType => {
+  const [dimensions, setDimensions] = useState(() => ({
```
Comment (andanf-e) [inline frontend/src/libs/hooks/use-get-window-dimensions/use-get-window-dimensions.hook.ts]:
```suggestion
  const [dimensions, setDimensions] = useState<WindowDimensions>(() => ({
```

File: frontend/src/libs/hooks/use-get-window-dimensions/use-get-window-dimensions.hook.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import { useEffect, useState } from '../hooks.js';
+
+type ReturnType = {
```
Comment (andanf-e) [inline frontend/src/libs/hooks/use-get-window-dimensions/use-get-window-dimensions.hook.ts]:
```suggestion
type WindowDimensions = {
```

File: frontend/src/pages/articles/components/articles-list/articles-list.tsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
+import { InfiniteScroll } from '~/libs/components/components.js';
+import { getArticleTags } from '~/libs/helpers/helpers.js';
+import { type ArticleWithCountsResponseDto } from '~/packages/articles/articles.js';
+
+import { ArticleCard } from '../components.js';
```
Comment (andanf-e) [inline frontend/src/pages/articles/components/articles-list/articles-list.tsx]: As it's the same folder as for articles-list
```suggestion
import { ArticleCard } from '../article-card/article-card.js';
```

**Comment by PablloZz** [comment]
image

**Comment by PablloZz** [comment]
image

**Comment by andanf-e** [comment]
> image

Now it looks narrower than tags displayed with prompts - maybe let's have it to have same with as a single item

**Review by andanf-e — DISMISSED** [review]

**Comment by PablloZz** [comment]
Do you mean make a check for the tags length and adding different styles depending on it?

**Comment by andanf-e** [comment]
> Do you mean make a check for the tags length and adding different styles depending on it?

No, to have a width of a single genre - save as of tag blocks when there are also prompt tags
image

Or it's already the same width?

**Comment by PablloZz** [comment]
> No, to have a width of a single genre - save as of tag blocks when there are also prompt tags

For now they are the same with, but due the 1fr, so when there's a single tag it expands.

**Comment by PablloZz** [comment]
Is it OK now?
image

**Comment by andanf-e** [comment]
> Is it OK now? image

Yep, thanks

File: frontend/src/pages/articles/components/articles-list/articles-list.tsx [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  onFetchData: () => void;
+  articles: ArticleWithCountsResponseDto[];
+};
+
+const ArticlesList: React.FC<Properties> = ({
+  hasMore,
+  articlesLength,
+  onFetchData,
+  articles,
+}) => {
```
Comment (mhrechkin) [inline frontend/src/pages/articles/components/articles-list/articles-list.tsx]:
```suggestion
}) => (
    <InfiniteScroll
      hasMore={hasMore}
      className={styles.articles}
      dataLength={articlesLength}
      fetchData={onFetchData}
    >
      {articles.map((article) => (
        <ArticleCard
          key={article.id}
          article={article}
          author={article.author}
          tags={getArticleTags(article)}
          reactions={article.reactions}
        />
      ))}
    </InfiniteScroll>
  );
```

File: frontend/src/pages/articles/components/empty-articles-placeholder/empty-articles-placeholder.tsx [hunk 6]
```diff
@@ -0,0 +1,14 @@
+import { Icon } from '~/libs/components/components.js';
+
+import styles from './styles.module.scss';
+
+const EmptyArticlesPlaceholder: React.FC = () => {
```
Comment (mhrechkin) [inline frontend/src/pages/articles/components/empty-articles-placeholder/empty-articles-placeholder.tsx]:
```suggestion
const EmptyArticlesPlaceholder: React.FC = () => (
    <h2 className={styles.emptyArticlesPlaceholder}>
      <Icon iconName="book" className={styles.emptyArticlesIcon} />
      It seems like there are no articles yet
    </h2>
  );
```

File: frontend/src/pages/articles/my-articles.tsx [hunk 7]
```diff
@@ -84,28 +112,41 @@ const MyArticles: React.FC = () => {
   return (
     <>
       <div className={styles.articlesWrapper}>
-        <InfiniteScroll
-          hasMore={hasMore}
-          className={styles.articles}
-          dataLength={articles.length}
-          fetchData={handleLoadArticles}
+        {articles.length ? (
```
Comment (mhrechkin) [inline frontend/src/pages/articles/my-articles.tsx]: Boolean(articles.length)
```suggestion
        {Boolean(articles.length) ? (
```

File: frontend/src/pages/profile/components/user-articles-genres-stats/libs/components/genres-list/genres-list.tsx [hunk 8]
```diff
@@ -0,0 +1,28 @@
+import { type UserArticlesGenreStatsItem } from '~/packages/users/users.js';
+
+import { GENRES_CHART_COLORS } from '../../constants/constants.js';
+import styles from './styles.module.scss';
+
+type Properties = {
+  chartData: UserArticlesGenreStatsItem[];
+};
+
+const GenresList: React.FC<Properties> = ({ chartData }) => {
```
Comment (mhrechkin) [inline frontend/src/pages/profile/components/user-articles-genres-stats/libs/components/genres-list/genres-list.tsx]:
```suggestion
const GenresList: React.FC<Properties> = ({ chartData }) => (
    <ul className={styles.list}>
      {chartData.map((data, index) => (
        <li
          key={data.key}
          className={styles.listItem}
          style={{
            color: GENRES_CHART_COLORS?.[index % GENRES_CHART_COLORS.length],
          }}
        >
          <span className={styles.itemName}>{data.name}</span>
        </li>
      ))}
    </ul>
  );
```

File: frontend/src/pages/profile/components/user-articles-genres-stats/libs/components/genres-list/genres-list.tsx [hunk 9]
```diff
@@ -0,0 +1,28 @@
+import { type UserArticlesGenreStatsItem } from '~/packages/users/users.js';
+
+import { GENRES_CHART_COLORS } from '../../constants/constants.js';
+import styles from './styles.module.scss';
+
+type Properties = {
+  chartData: UserArticlesGenreStatsItem[];
```
Comment (mhrechkin) [inline frontend/src/pages/profile/components/user-articles-genres-stats/libs/components/genres-list/genres-list.tsx]: B18

**Review by mhrechkin — CHANGES_REQUESTED** [review]

File: frontend/src/pages/profile/components/user-articles-genres-stats/libs/components/genres-list/genres-list.tsx (hunk 9 above)
Comment (PablloZz) [reply frontend/src/pages/profile/components/user-articles-genres-stats/libs/components/genres-list/genres-list.tsx:7]: But the word "data" is already plural form

File: frontend/src/pages/articles/my-articles.tsx (hunk 7 above)
Comment (PablloZz) [reply frontend/src/pages/articles/my-articles.tsx]: Can I change eslint configuration?
image

File: frontend/src/pages/profile/components/user-articles-genres-stats/libs/components/genres-list/genres-list.tsx (hunk 9 above)
Comment (PablloZz) [reply frontend/src/pages/profile/components/user-articles-genres-stats/libs/components/genres-list/genres-list.tsx]: but the word "data" already has a plural form

Comment (andanf-e) [reply frontend/src/pages/profile/components/user-articles-genres-stats/libs/components/genres-list/genres-list.tsx]: Maybe let's call this `genresStatistics` - just to make it more precise what does it mean

**Review by andanf-e — DISMISSED** [review]

File: frontend/src/pages/articles/my-articles.tsx (hunk 7 above)
Comment (andanf-e) [reply frontend/src/pages/articles/my-articles.tsx]: Yes

**Review by andanf-e — DISMISSED** [review]

**Review by mhrechkin — DISMISSED** [review]

**Review by andanf-e — APPROVED** [review]

**Pull request merged**